api/jobs.yaml

```diff
@@ -398,7 +398,7 @@ definitions:
           matches to each of these labels.
       id:
         type: string
-        pattern: ^[a-zA-Z][a-zA-Z0-9_-]*$
+        pattern: ^[a-zA-Z0-9][a-zA-Z0-9_-]+$
```

Comment:
I didn't think these IDs could include underscores (I do recognize that this predated your PR)

Reply:
Since this regex is used by both dsub and cromwell backend, do we want to have a reviewer from the dsub side to take a look?

Reply:
The end result here should be a pattern that matches all possible dsub or cromwell IDs. I think this satisfies it (changing the * --> + should be fine since I'm pretty confident we have a minimum id length).